### PR TITLE
Curated Cards : Coveo Query Formation and Code cleanup

### PR DESCRIPTION
**Pull request by noormd08**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID:

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live/en/curated-cards
- After: https://feature-exlm-596-coveo-fields--exlm--adobe-experience-league.hlx.live/en/curated-cards

image

### CODE REVIEW

File: blocks/curated-cards/curated-cards.js [hunk 1]
```diff
@@ -12,18 +13,41 @@ export default async function decorate(block) {
...
+  const role = block.querySelector('div:nth-child(6) > div')?.textContent?.trim()?.toLowerCase();
+  const level = block.querySelector('div:nth-child(7) > div')?.textContent?.trim()?.toLowerCase();
+  const sortBy = block.querySelector('div:nth-child(9) > div')?.textContent?.trim()?.toLowerCase();
+  const sortCriteria = SORT_OPTIONS[sortBy?.toUpperCase()];
   const noOfResults = 4;
+  const productKey = 'exl:solution';
+  const featureKey = 'exl:feature';
+  const capability = {};
+  const regex = /(?:exl:solution\/|exl:feature\/)([^,]+)/g;
+  const matches = capabilityElement.match(regex);
```
Comment (Sivaramvt) [inline blocks/curated-cards/curated-cards.js:27]: Use string.startswith to determine if the given tag is solution or feature

File: scripts/browse-card/browse-cards-constants.js [hunk 2]
```diff
@@ -38,4 +38,8 @@ const CONTENT_TYPES = Object.freeze({
   },
 });
 
-export default CONTENT_TYPES;
+export const SORT_OPTIONS = Object.freeze({
```
Comment (Sivaramvt) [inline scripts/browse-card/browse-cards-constants.js:41]: Pls make it COVEO_SORT_OPTIONS

**Review by Sivaramvt — DISMISSED** [review]

**Comment by Sivaramvt** [comment]
Pending items

1. Thumbnail
2. CSS - Reusing browse-cards-block
3. Constant / SORT renaming
4. Block JS - Tag matching
5. Placeholder

**Comment by noormd08** [comment]
> Pending items
> 
> 1. Thumbnail - **Done**
> 2. CSS - Reusing browse-cards-block - **Will be taken as a part of different PR (By Neha)**
> 3. Constant / SORT renaming - **Done**
> 4. Block JS - Tag matching - **Will be taken as a part of different PR (By Noor and Ruhi)**
> 5. Placeholder - **Will be taken as a part of different PR (By Neha)**

**Review by tprasad05 — APPROVED** [review]

**Pull request merged**